### DISCUSSION TITLE
Costructing a route using intermediates difficult

### DISCUSSION THREAD

**Discussion by ManDay** [osmandapp/OsmAnd#11599] [Ideas]
It's not exactly a bug, although there is a bug which could help with that. I find it rather difficult to "construct a route", that is to say, given a start an a destination, add intermediates such that resulting route matches what you want to achieve.

I suppose what is lacking in that regard (more of a feature request here), is that you could add intermediates anywhere in the current list of waypoints, rather than just at the very end (or, from the marker context menu under "Directions", also as the very first intermediate). Currently, you and up adding intermediates at the end and then have to reorder them.

What makes reodering particularly difficult (and hence the **bug**), is that, while the list of current intermediates *wants* to show a label "Intermediate point No. 123" for each intermediate (right next to the distance), that text is actually cut short on my device such that I only see "Intermediate po…", which kind of defies the idea. I suppose on larger screens or with smaller text, this fits in nicely, but perhaps this could be shortned by a bit (e.g. "Interm. 123") to work reliably on more devices.

**Comment by chau-intl** [comment]
I agree on the difficulties in redefining the route after the first and the last intermediate destination has been used.

I think a simple approach would be to:
1. User long press on a point on/very near to an already calculated route.
2. User selects Directions in the lower right corner.
3. An extra entry pops up in the menu saying something like "Add as intermediate destination".
4. The app determines where the point is located on the route with respect to the existing waypoints.
5. The route is recalculated.

After this the user has the ability to move the intermediate destination. I think it sounds so simple that I'm probably unaware of all the complexities involved :)

I do however miss this feature almost everytime I plan a route.
